test/__snapshots__/test.js.snap

```diff
@@ -1587,46 +1907,141 @@ const Impl = require(\\"../implementations/MixedIn.js\\");
-};
 
-exports.create = function create(globalObject, constructorArgs, privateData) {
-  if (globalObject[ctorRegistry] === undefined) {
-    throw new Error(\\"Internal error: invalid global object\\");
-  }
+  const { wrapper } = {
+    wrapper(...args) {
+      let completion;
+      let thisArg = this;
+      let O = value;
+      let X = O;
 
-  const ctor = globalObject[ctorRegistry][\\"Overloads\\"];
-  if (ctor === undefined) {
-    throw new Error(\\"Internal error: constructor Overloads is not installed on the passed global object\\");
-  }
+      if (typeof O !== \\"function\\") {
+        X = O[\\"acceptNode\\"];
+        if (typeof X !== \\"function\\") {
+          throw new TypeError(\`\${context} does not correctly implement NodeFilter.\`);
+        }
+        thisArg = O;
+      }
 
-  let obj = Object.create(ctor.prototype);
-  obj = exports.setup(obj, globalObject, constructorArgs, privateData);
-  return obj;
-};
-exports.createImpl = function createImpl(globalObject, constructorArgs, privateData) {
+      for (let i = 0; i < args.length; i++) {
+        args[i] = utils.tryWrapperForImpl(args[i]);
+      }
+      if (args.length < 1) {
+        for (let i = args.length; i < 1; i++) {
+          args[i] = undefined;
+        }
+      } else if (args.length > 1) {
+        args.length = 1;
+      }
+
+      let callResult = Reflect.apply(X, thisArg, args);
+
+      callResult = conversions[\\"unsigned short\\"](callResult, { context: context });
+
+      return callResult;
+    }
+  };
+
+  Object.defineProperties(wrapper, {
+    [utils.wrapperSymbol]: { value },
+    value: { value, enumerable: true }
+  });
+
+  return wrapper;
+};
+
+exports.install = function install(globalObject) {
+  const NodeFilter = () => {
+    throw new TypeError(\\"Illegal invocation\\");
+  };
+
+  Object.defineProperties(NodeFilter, {
+    FILTER_ACCEPT: { value: 1, enumerable: true },
+    FILTER_REJECT: { value: 2, enumerable: true },
+    FILTER_SKIP: { value: 3, enumerable: true },
+    SHOW_ALL: { value: 0xffffffff, enumerable: true },
+    SHOW_ELEMENT: { value: 0x1, enumerable: true },
+    SHOW_ATTRIBUTE: { value: 0x2, enumerable: true },
+    SHOW_TEXT: { value: 0x4, enumerable: true },
+    SHOW_CDATA_SECTION: { value: 0x8, enumerable: true },
+    SHOW_ENTITY_REFERENCE: { value: 0x10, enumerable: true },
+    SHOW_ENTITY: { value: 0x20, enumerable: true },
+    SHOW_PROCESSING_INSTRUCTION: { value: 0x40, enumerable: true },
+    SHOW_COMMENT: { value: 0x80, enumerable: true },
+    SHOW_DOCUMENT: { value: 0x100, enumerable: true },
+    SHOW_DOCUMENT_TYPE: { value: 0x200, enumerable: true },
+    SHOW_DOCUMENT_FRAGMENT: { value: 0x400, enumerable: true },
+    SHOW_NOTATION: { value: 0x800, enumerable: true }
+  });
+
+  if (globalObject[ctorRegistry] === undefined) {
+    globalObject[ctorRegistry] = Object.create(null);
+  }
+  globalObject[ctorRegistry][interfaceName] = NodeFilter;
```

Comment:
Does it belong in the ctorRegistry? Maybe ctorRegistry is misnamed?

Reply:
Probably not.